src/eradius_proxy.erl

```diff
@@ -39,25 +50,25 @@
                           {timeout, ?DEFAULT_TIMEOUT},
                           {retries, ?DEFAULT_RETRIES}]).
 
--type route() :: eradius_client:nas_address().
--type routes() :: [{Name :: string(), route()}].
+-type route() :: {default_route, eradius_client:nas_address()}.
+-type routes() :: {routes, [{Name :: string(), route()}]}.
 -type undefined_route() :: {undefined, 0, []}.
 
 radius_request(Request, _NasProp, Args) ->
-    DefaultRoute = proplists:get_value(default_route, Args),
+    DefaultRoute = get_proxy_opt(default_route, Args),
+    Routes = get_proxy_opt(routes, Args),
     Options = proplists:get_value(options, Args, ?DEFAULT_OPTIONS),
     Username = eradius_lib:get_attr(Request, ?User_Name),
-    Routes = proplists:get_value(routes, Args, []),
     {NewUsername, Route} = resolve_routes(Username, DefaultRoute, Routes, Options),
     Retries = proplists:get_value(retries, Options, ?DEFAULT_RETRIES),
     Timeout = proplists:get_value(timeout, Options, ?DEFAULT_TIMEOUT),
     SendOpts = [{retries, Retries}, {timeout, Timeout}],
     send_to_server(new_request(Request, Username, NewUsername), Route, SendOpts).
 
 validate_arguments(Args) ->
-    DefaultRoute = proplists:get_value(default_route, Args, {undefined, 0, []}),
+    DefaultRoute = lists:keyfind(default_route, 1, Args),
```

Comment:
`lists:keyfind/3` will return `false` - this is expected as default value for `DefaultRoute` for `validate_route/1`?

Reply:
this is reworked as well